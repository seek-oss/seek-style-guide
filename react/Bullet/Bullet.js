import React from 'react';

import Text from '../Text/Text';

export default function Bullet(props) {
  return <Text bullet {...props} />;
}
