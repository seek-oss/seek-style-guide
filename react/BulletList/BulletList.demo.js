import React from 'react';
import { BulletList, Bullet } from 'seek-style-guide/react';

const renderChildren = (size = 'Default') =>
  size === 'Default' ? (
    <BulletList>
      <Bullet>Bullet</Bullet>
      <Bullet>Bullet</Bullet>
      <Bullet>Bullet</Bullet>
    </BulletList>
  ) : (
    <BulletList>
      <Bullet {...{ [size.toLowerCase()]: true }}>Bullet</Bullet>
      <Bullet {...{ [size.toLowerCase()]: true }}>Bullet</Bullet>
      <Bullet {...{ [size.toLowerCase()]: true }}>Bullet</Bullet>
    </BulletList>
  );

export default {
  route: '/bullet-list',
  title: 'Bullet List',
  category: 'Typography',
  component: 'div',
  initialProps: {
    children: renderChildren()
  },
  options: [
    {
      label: 'Size',
      type: 'radio',
      states: ['Default', 'Large', 'Standard', 'Small'].map(size => ({
        label: `${String(size)}`,
        transformProps: props => ({
          ...props,
          children: renderChildren(size)
        })
      }))
    }
  ]
};
