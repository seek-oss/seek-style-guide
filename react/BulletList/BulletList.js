import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './BulletList.less';

export default function BulletList({ className, children }) {
  return <ul className={classNames(styles.root, className)}>{children}</ul>;
}

BulletList.displayName = 'BulletList';

BulletList.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
