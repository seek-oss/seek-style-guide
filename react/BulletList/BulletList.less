@import (reference) "~seek-style-guide/theme";

.root {
  margin-bottom: 2 * @row-height;
}
