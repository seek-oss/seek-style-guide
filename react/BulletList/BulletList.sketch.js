import React from 'react';
import { BulletList, Bullet } from 'seek-style-guide/react';
export const symbols = {
  'BulletList/Small': (
    <BulletList>
      <Bullet small>Bullet point 1</Bullet>
      <Bullet small>Bullet point 2</Bullet>
      <Bullet small>Bullet point 3</Bullet>
    </BulletList>
  ),
  'BulletList/Standard': (
    <BulletList>
      <Bullet>Bullet point 1</Bullet>
      <Bullet>Bullet point 2</Bullet>
      <Bullet>Bullet point 3</Bullet>
    </BulletList>
  ),
  'BulletList/Subheading': (
    <BulletList>
      <Bullet subheading>Bullet point 1</Bullet>
      <Bullet subheading>Bullet point 2</Bullet>
      <Bullet subheading>Bullet point 3</Bullet>
    </BulletList>
  )
};
