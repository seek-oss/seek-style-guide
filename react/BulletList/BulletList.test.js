import React from 'react';
import { shallow } from 'enzyme';

import BulletList from './BulletList';

import Bullet from '../Bullet/Bullet';

describe('BulletList', () => {
  it('renders with standard text', () => {
    const wrapper = shallow(
      <BulletList>
        <Bullet>Foo</Bullet>
        <Bullet>Bar</Bullet>
      </BulletList>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
