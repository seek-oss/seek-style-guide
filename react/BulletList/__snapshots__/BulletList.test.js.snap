// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletList renders with standard text 1`] = `
<ul
  className="BulletList__root"
>
  <Bullet>
    Foo
  </Bullet>
  <Bullet>
    Bar
  </Bullet>
</ul>
`;
